source/ceylon/uri/parsing.ceylon

```diff
@@ -128,13 +125,9 @@ shared Uri parse(String uri) {
         return remains;
     }
 
-    {Parameter*} parsePathSegmentParameters(String part) {
-        value list = LinkedList<Parameter>();
-        for(param in part.split((Character ch) => ch == ';', true, false)) {
-            list.add(parseParameter(param));
-        }
-        return list;
-    }
+    {Parameter*} parsePathSegmentParameters(String part)
+        =>  [ for(param in part.split((Character ch) => ch == ';', true, false))
```

Comment:
Good point, I should have at least changed it to `';'.equals`. But overall I was going for a minimal diff.